larray/tests/test_array.py

```diff
@@ -263,13 +263,21 @@ def test_getitem(self):
         res = arr[arr.b['l1']]
         assert_array_equal(res, arr.data[:, 0])
 
+        # scalar group on another axis
+        arr = ndtest((3, 2))
+        alt_a = Axis("alt_a=a1..a2")
+        lgroup = alt_a['a1']
+        assert_array_equal(arr[lgroup], arr['a1'])
+        pgroup = alt_a.i[0]
+        assert_array_equal(arr[pgroup], arr['a1'])
+
         # key with duplicate axes
         with self.assertRaises(ValueError):
             la[age[1, 2], age[3, 4]]
 
         # key with invalid axis
         bad = Axis(3, 'bad')
-        with self.assertRaises(KeyError):
+        with self.assertRaises(ValueError):
```

Comment:
The code below (use of an invalid axis) throws a `ValueError` from `_translated_key` at line 2027 corresponding to the case of "duplicate axes". This is weird and wrong. Oops.

Reply:
I think that in `array.__getitem__`, the exception should be a KeyError, but this **might** be (I have NOT thought this through) be a case where it makes more sense to throw a ValueError in _translated_key then catch it somewhere and reraise KeyError instead. Or it is a case where we need to raise KeyError in all cases. I don't know. Depends where _translated_key is used, which I don't remember right now.

Reply:
As I understand issue #178, you want to allow users to make a selection by passing a group from another axis. Meaning we don't care about the name of the axis associated with the group. Meaning passing an invalid axis will make `_translate_axis_key_chunk` to execute `axis_key = axis_key.to_label()` and to search for a compatible below leading to possible duplicate axis.

Reply:
Indeed. I finally understand the change, now. Sorry for all the fuzz. The comments surrounding the test should be changed though, because what we test is not really the presence of an "invalid" axis, it is just a roundabout way to get a duplicate axis (which is an interesting test too).